packages/dashboard-message-bus/lib/DashboardMessageBus.ts

```diff
@@ -0,0 +1,184 @@
+import WebSocket from "isomorphic-ws";
+import delay from "delay";
+import { EventEmitter } from "events";
+import {
+  base64ToJson,
+  broadcastAndAwaitFirst,
+  broadcastAndDisregard,
+  createMessage,
+  jsonToBase64,
+  startWebSocketServer
+} from "./utils";
+import { promisify } from "util";
+
+interface UnfulfilledRequest {
+  socket: WebSocket;
+  data: WebSocket.Data;
+}
+
+export class DashboardMessageBus extends EventEmitter {
+  private requestsServer: WebSocket.Server;
+  private listenServer: WebSocket.Server;
```

Comment:
The naming here is a bit asymmetric - are there other names we could use?

Reply:
Yeah most of the property names in that entire class are a bit off, but I'm not sure what the way to go is.

Reply:
Updated to pub-sub terminology.